LoginDemo/LearningSelenium/SingletonTest.cs

```diff
@@ -0,0 +1,101 @@
+﻿using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium;
+using LearningSelenium.Singletons;
+using LearningSelenium.Utilities;
+using OpenQA.Selenium.Support.UI;
+
+namespace LearningSelenium
+{
+    public class SingletonTest
+    {
+        WebDriver driver;
+        Utility utilityObj;
+
+        [SetUp]
+        public void Setup()
+        {
+            driver = new ChromeDriver();
+            //driver.Manage().Window.Maximize();
+            utilityObj = Utility.GetInstance(driver);
+            utilityObj.SetImplicitWait(20);
+
+        }
+
+        [Test]
+        public void TestLoginFail()
+        {
+            Singleton singletonObj = Singleton.GetInstance(driver);
+            driver.Navigate().GoToUrl("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
+            utilityObj.SetImplicitWait(20);
+            singletonObj.EnterUserName("admin");
+            singletonObj.EnterPassword("admin1234");
+            singletonObj.LoginClick();
+            utilityObj.SetExplicitWait(singletonObj.GetErrorMessage(),20);
+            Assert.That(singletonObj.GetErrorMessage(),Is.EqualTo("Invalid credentials") , "username or password is incorrect");
+        }
+
+        [Test]
+        public void TestLoginSuccess()
+        {
+            Singleton singletonObj = Singleton.GetInstance(driver);
+            driver.Navigate().GoToUrl("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
+            utilityObj.SetImplicitWait(20);
+            singletonObj.EnterUserName("Admin");
+            singletonObj.EnterPassword("admin123");
+            singletonObj.LoginClick();
+            utilityObj.SetExplicitWait(singletonObj.GetSuccessMessage(), 20);
+            Assert.That(singletonObj.GetSuccessMessage(), Is.EqualTo("Dashboard"), "Dashboard not loading...");
+
+        }
+
+        [Test]
+        public void TestLogin()
+        {
+            Singleton singletonObj = Singleton.GetInstance(driver);
+            driver.Navigate().GoToUrl("https://www.saucedemo.com/"); 
+            utilityObj.SetImplicitWait(20);
+            singletonObj.EnterSwagUserName("standard_user");
+            singletonObj.EnterSwagPassword("secret_sauce");
+            singletonObj.SwagLoginClick();
+            utilityObj.SetExplicitWait(singletonObj.GetSwagSuccessMessage(), 20);
+            Assert.AreEqual("Products", singletonObj.GetSwagSuccessMessage(), "loged user name is mismatch");
+        }
+
+        [Test]
+        public void TestProduct()
+        {
+            TestLogin();
+            Product productObj = Product.GetInstance(driver);
+            productObj.GetItem1();
+            productObj.LinkItem1();
+            driver.FindElement(productObj.backMenu).Click();
+            productObj.GetItem2();
+            productObj.LinkItem2();
+            driver.FindElement(productObj.backMenu).Click();
+            utilityObj.SetExplicitWait(productObj.GetItemCount().ToString(), 20);
```

Comment:
Next step:
Learn about  Factory Method pattern.
Read about test data and config data files: Test data - Test data - kind of data, that test is built one, like name of a game, country in registration, etc.

Config data - Config data - kind of data that is used to configure a project in whole (URL, browser name, wait time, etc.):

How to store data - Data should be stored separately (config and test) in xml or json file (chosen format should be used in a whole project).

Reading data - For reading config and test data separate utility class should be implemented;

Data - Any number different from 0 and 1 if it's used in a project should stored as a constant, if it's not suitable to be took out to test/config data

Reply:
Factory Method, test data config data will be implemented in next task